src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -45,30 +46,57 @@ function isValidFileType(fileName) {
   return FILE_TYPES.some(type => fileName.toLowerCase().endsWith(type));
 }
 
-function isValidImageSize(result) {
+// data urls start with data:image/jpg; and we need to pull out the
+// image/jpg part so canvas doesn't change the file type
+function getMimeType(dataUrl) {
+  // limit to 2 so we don't try to split a giant data url string
+  const prefix = dataUrl.split(';', 2)[0];
+  return prefix.replace('data:', '');
+}
+
+// If any of the image dimensions are greater than the max specified, 
+// resize it down to that dimension while keeping the aspect ratio
+// intact
+function resizeToMaxDimension(img, mimeType, maxDimension) {
```

Comment:
This is really helpful for the user. Should we default to the full canvas area or the cropping frame? Also, should we use "default" rather than "max", as users are still able to zoom in?

Reply:
Might be helpful to call this `maybeResizeToMaxDimensions` to the name of this function since it doesn't upscale the image to the max size.

I also find using an object parameter makes calls to this function easier to read when there are more than 2 arguments  e.g. `({img, mimeType, maxDimension})`

Reply:
Yeah, I'll make the name clearer